File: src/main/java/com/googongill/aditory/controller/dto/link/LinkDetailResponse.java
package com.googongill.aditory.controller.dto.link;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.googongill.aditory.domain.Link;
import lombok.Builder;

import java.time.LocalDateTime;

@Builder
@JsonSerialize
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public class LinkDetailResponse {
    private Long linkId;
    private String title;
    private String summary;
    private Boolean linkState;
    private String url;
    private Long categoryId;
    private String categoryName;
    private LocalDateTime createdAt;
    private LocalDateTime lastModifiedAt;

    public static LinkDetailResponse of(Link link) {
        return LinkDetailResponse.builder()
                .linkId(link.getId())
                .title(link.getTitle())
                .summary(link.getSummary())
                .linkState(link.getLinkState())
                .url(link.getUrl())
                .categoryId(link.getCategory().getId())
                .categoryName(link.getCategory().getCategoryName())
                .createdAt(link.getCreatedAt())
                .lastModifiedAt(link.getLastModifiedAt())
                .build();
    }
}

File: src/main/java/com/googongill/aditory/controller/dto/link/response/LinkDetailResponse.kt
package com.googongill.aditory.controller.dto.link.response

import com.fasterxml.jackson.annotation.JsonAutoDetect
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.googongill.aditory.domain.Link
import java.time.LocalDateTime

@JsonSerialize
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
data class LinkDetailResponse(
    val linkId: Long?,
    val title: String,
    val summary: String?,
    val url: String,
    val linkState: Boolean,
    val categoryId: Long?,
    val categoryName: String,
    val createdAt: LocalDateTime?,
    val lastModifiedAt: LocalDateTime?
) {

    companion object {
        fun of(link: Link): LinkDetailResponse {
            return LinkDetailResponse(
                linkId = link.id,
                title = link.title,
                summary = link.summary,
                linkState = link.linkState,
                url = link.url,
                categoryId = link.category.id,
                categoryName = link.category.categoryName,
                createdAt = link.createdAt,
                lastModifiedAt = link.lastModifiedAt
            )
        }
    }

}
